### ISSUE TITLE
Detection of energy mode in `Universe.plot`

### ISSUE THREAD

**Issue by pshriwise** [openmc-dev/openmc#2064]
The `Universe.plot` method was updated in #2019 to export XML files and use the C++ API for generating a pixel map of IDs, which is much faster. This method fails when multi-group materials are present, however, with the message:

```shell
RuntimeError: Macroscopic can not be used in continuous-energy mode.
```

So some extra logic is needed to detect the types of materials applied in the model.

**Comment by pshriwise** [comment]
Closes #2064.

**Comment by paulromano** [comment]
Closed by #2145
